Только в socketnotifier: qabstractsocketengine.cpp
Только в socketnotifier: qabstractsocketengine_p.h
Только в socketnotifier: qnativesocketengine.cpp
Только в socketnotifier: qnativesocketengine_p.h
Только в socketnotifier: qnativesocketengine_unix.cpp
diff -uwd qsocketnotifier/qsocketnotifier.pro socketnotifier/qsocketnotifier.pro
--- qsocketnotifier/qsocketnotifier.pro	2012-10-22 10:52:26.000000000 +0300
+++ socketnotifier/qsocketnotifier.pro	2012-11-16 07:36:28.477549631 +0200
@@ -1,10 +1,12 @@
 CONFIG += testcase
-CONFIG += parallel_test
 TARGET = tst_qsocketnotifier
-QT = core-private network-private testlib
-SOURCES = tst_qsocketnotifier.cpp
+QT       = core-private network-private testlib network
+SOURCES  = tst_qsocketnotifier.cpp qabstractsocketengine.cpp qnativesocketengine.cpp qnativesocketengine_unix.cpp
+HEADERS  = qabstractsocketengine_p.h qnativesocketengine_p.h
+DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
 
-requires(contains(QT_CONFIG,private_tests))
+QNETWORK_SRC = $$QT_SOURCE_TREE/src/network
+INCLUDEPATH += $$QNETWORK_SRC
 
-include(../../../network/socket/platformsocketengine/platformsocketengine.pri)
-DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
+DESTDIR  = ../
+include(../common.pri)
diff -uwd qsocketnotifier/tst_qsocketnotifier.cpp socketnotifier/tst_qsocketnotifier.cpp
--- qsocketnotifier/tst_qsocketnotifier.cpp	2012-10-22 10:48:47.000000000 +0300
+++ socketnotifier/tst_qsocketnotifier.cpp	2012-11-16 07:39:10.358452754 +0200
@@ -50,16 +50,11 @@
 #include <QtNetwork/QTcpSocket>
 #include <private/qnativesocketengine_p.h>
 #define NATIVESOCKETENGINE QNativeSocketEngine
-#ifdef Q_OS_UNIX
 #include <private/qnet_unix_p.h>
 #include <sys/select.h>
-#endif
 #include <limits>
 
-#if defined (Q_CC_MSVC) && defined(max)
-#  undef max
-#  undef min
-#endif // Q_CC_MSVC
+#include "eventdispatcher_libevent.h"
 
 class tst_QSocketNotifier : public QObject
 {
@@ -302,5 +297,12 @@
 #endif
 }
 
-QTEST_MAIN(tst_QSocketNotifier)
 #include <tst_qsocketnotifier.moc>
+
+int main(int argc, char** argv)
+{
+    QCoreApplication::setEventDispatcher(new EventDispatcherLibEvent);
+    QCoreApplication app(argc, argv);
+    tst_QSocketNotifier t;
+    return QTest::qExec(&t, argc, argv);
+}
